src/ssl.c

```diff
@@ -23670,11 +23670,45 @@ long wolfSSL_set_tlsext_debug_arg(WOLFSSL* ssl, void *arg)
 #endif /* HAVE_PK_CALLBACKS */
 
 #if defined(OPENSSL_ALL) || defined(WOLFSSL_HAPROXY)
-const unsigned char *SSL_SESSION_get0_id_context(const WOLFSSL_SESSION *sess, unsigned int *sid_ctx_length)
+WOLFSSL_API const unsigned char *wolfSSL_SESSION_get0_id_context(
+                      const WOLFSSL_SESSION *sess, unsigned int *sid_ctx_length)
 {
     sess = ClientSessionToSession(sess);
     return wolfSSL_SESSION_get_id((WOLFSSL_SESSION *)sess, sid_ctx_length);
 }
+WOLFSSL_API int wolfSSL_SESSION_set1_id(WOLFSSL_SESSION *s,
+                                 const unsigned char *sid, unsigned int sid_len)
+{
+    if (s == NULL) {
+        return WOLFSSL_FAILURE;
+    }
+    if (sid_len > SSL_MAX_SSL_SESSION_ID_LENGTH) {
+        return WOLFSSL_FAILURE;
+    }
+    s->sessionIDSz = sid_len;
+    if (sid != s->sessionID) {
+        XMEMCPY(s->sessionID, sid, sid_len);
+    }
+    return WOLFSSL_SUCCESS;
+}
+
+WOLFSSL_API int wolfSSL_SESSION_set1_id_context(WOLFSSL_SESSION *s,
+                         const unsigned char *sid_ctx, unsigned int sid_ctx_len)
+{
+    if (s == NULL) {
+        return WOLFSSL_FAILURE;
+    }
+    if (sid_ctx_len > SSL_MAX_SSL_SESSION_ID_LENGTH) {
+        return WOLFSSL_FAILURE;
+    }
+    s->sessionCtxSz = sid_ctx_len;
+    if (sid_ctx != s->sessionCtx) {
+        XMEMCPY(s->sessionCtx, sid_ctx, sid_ctx_len);
```

Comment:
Seems odd that `sessionCtx` uses `ID_LEN`: `byte               sessionCtx[ID_LEN]; /* app specific context id */`
These functions are using `SSL_MAX_SSL_SESSION_ID_LENGTH` and both are 32-bytes, but would prefer using the same macro (or refactoring to use `sizeof()`.

Reply:
@dgarske That makes sense. Updated to use existing `ID_LEN` and removed `WOLFSSL_API` from `ssl.c`.  BTW, `SSL_MAX_SSL_SESSION_ID_LENGTH` was defined but unused and I decided to use the same macro for both functions since they are both defined as 32 bytes in Openssl:
```
 # define SSL_MAX_SSL_SESSION_ID_LENGTH           32
 # define SSL_MAX_SID_CTX_LENGTH                  32
```
I think `SSL_MAX_SSL_SESSION_ID_LENGTH` could be removed now unless any of our third-party-supported libraries use it.

Reply:
`SSL_MAX_SSL_SESSION_ID_LENGTH` is for openssl compatibility. I think it should be `#define SSL_MAX_SSL_SESSION_ID_LENGTH ID_LEN` though. Feel free to update that.